enclave-tls/samples/enclave-tls-client/client.c

```diff
@@ -38,20 +39,49 @@ static sgx_enclave_id_t load_enclave(bool debug_enclave)
 	int updated = 0;
 	int ret = sgx_create_enclave(ENCLAVE_FILENAME, debug_enclave, &t, &updated, &eid, NULL);
 	if (ret != SGX_SUCCESS) {
-		ETLS_ERR("Failed to load enclave %d\n", ret);
-		return -1;
+		printf("Failed to load enclave %d\n", ret);
+		return 0;
 	}
 
-	ETLS_DEBUG("Success to load enclave id %ld\n", eid);
+	printf("Success to load enclave id %ld\n", eid);
 
 	return eid;
 }
-#endif
 // clang-format on
 
-int enclave_tls_echo(int fd, enclave_tls_log_level_t log_level, char *attester_type,
-		     char *verifier_type, char *tls_type, char *crypto_type, bool mutual,
-		     bool debug_enclave)
+int enclave_tls_client_startup(enclave_tls_log_level_t log_level, char *attester_type,
+			       char *verifier_type, char *tls_type, char *crypto_type, bool mutual,
+			       bool debug_enclave, char *ip, int port)
+{
+	sgx_enclave_id_t enclave_id = 0;
+	unsigned long flags = 0;
+	uint32_t s_ip = inet_addr(ip);
+	uint16_t s_port = htons((uint16_t)port);
+
+	enclave_id = load_enclave(debug_enclave);
+	if (enclave_id == 0) {
+		printf("Failed to load sgx stub enclave\n");
+		return -1;
+	}
+	if (mutual)
+		flags |= ENCLAVE_TLS_CONF_FLAGS_MUTUAL;
+
+	int ret = 0;
+	int sgx_status = ecall_etls_client_startup((sgx_enclave_id_t)enclave_id, &ret,
+						   (sgx_enclave_id_t)enclave_id, log_level,
+						   attester_type, verifier_type, tls_type,
+						   crypto_type, flags, s_ip, s_port);
+	if (sgx_status != SGX_SUCCESS || ret)
+		printf("failed to startup client: sgx status %#x return %#x\n", sgx_status, ret);
```

Comment:
Use ETLS_ERR()?

Reply:
In client, not running in SGX enclave, it should use printf. Because ETLS_* is defined in trusted zone.

Reply:
The original implementation already considers this:
```C
#ifdef OCCLUM
  #define FPRINTF(io, fmt, ...)                  \
        do {                                     \
                fprintf(io, fmt, ##__VA_ARGS__); \
                fflush(io);                      \
        } while (0)
#else
  #define FPRINTF(io, fmt, ...)                  \
        do {                                     \
                fprintf(io, fmt, ##__VA_ARGS__); \
        } while (0)
#endif
```

We hope the applications based on enclave-tls can use ETLS_* log functions even it runs under non-sgx platform.